packages/fiber/src/core/renderer.ts

```diff
@@ -263,36 +263,50 @@ function createRenderer<TCanvas>(roots: Map<TCanvas, Root>, getEventPriority?: (
     })
   }
 
-  const reconciler = Reconciler({
+  const reconciler = Reconciler<
+    string,
+    InstanceProps,
+    UseBoundStore<RootState>,
+    Instance,
+    never,
+    Instance,
+    Instance,
+    Instance,
+    never,
+    Array<boolean | number | DiffSet>,
+    never,
+    typeof setTimeout | undefined,
+    -1
+  >({
     createInstance,
     removeChild,
     appendChild,
     appendInitialChild: appendChild,
     insertBefore,
-    supportsMicrotask: true,
-    warnsIfNotActing: true,
     supportsMutation: true,
     isPrimaryRenderer: false,
+    supportsPersistence: true,
+    supportsHydration: true,
```

Comment:
We'll want both of these as `false` here. Immutable trees cannot be mutated and are exclusive with the `supportsMutation` option which better represents a threejs scene graph (see [reconciler modes](https://github.com/facebook/react/tree/main/packages/react-reconciler#modes)).